development process updates

edublancas opened the pull request:
updated the contribution guidelines based on feedback I've heard from the team

<!-- readthedocs-preview ploomber-contributing start -->
----
:books: Documentation preview :books:: https://ploomber-contributing--82.org.readthedocs.build/en/82/

<!-- readthedocs-preview ploomber-contributing end -->

neelasha23 commented on contributing/process-outline.md:
````diff
@@ -34,26 +33,35 @@ you can choose one that you're interested in working on.
+For more information, see [](submitting-pr.md).
+```
+
+- Once you've completed the work, submit a draft pull request to the repository. 
````
We can mention here that `draft` is when it's still work in progress. Need to mark `ready for review` when work is completed.

edublancas replied:
fixed

neelasha23 replied:
Few typos here:
`opem`, `clicck`

neelasha23 commented on contributing/review.md:
```diff
@@ -1,7 +1,39 @@
 # Reviewing PRs
 
-## Guidelines
+## The first review should be throughout
```
Did you mean `thorough`?

neelasha23 commented on contributing/review.md:
```diff
@@ -1,7 +1,39 @@
+
+## You're responsible for resolving conversation
+
+The reviewer is responsible for clicking on `Resolve conversation`. Contributors should address observations but it's up to the reviewer to mark them as resolved (as they're the ones that verify that their observations were effectively addressed).
```
Maybe contributor can add a comment tagging the reviewer once all comments have been resolved.

edublancas replied:
I think it's better to request another review (added a sentence to be more explicit)

neelasha23 commented on contributing/submitting-pr.md:
````diff
@@ -90,6 +90,47 @@ Keep in mind this guidelines when writing changelog messages:
+    assert str(excinfo).value == message
+```
+
+Of course, the test won't pass since you haven't implemetned such validation, but you can now go, implement it and verify that the test passes.
````
`implemetned` - typo

edublancas replied:
fixed

neelasha23 replied:
I still see `implemetned`

edublancas replied:
maybe github  doesn't update the view? this is the commit that I added, you can see I fixed the typo: https://github.com/ploomber/contributing/pull/82/commits/a6895752190de3974ee6cf74d88f1f7ae1de749b

neelasha23 commented on contributing/submitting-pr.md:
```diff
@@ -202,7 +247,49 @@ Then ensure the menu displays the original repository (`ploomber/PROJECT_NAME`)
+Then click on open **Draft Pull Request.** Opening a PR will trigger building
+documentation and running the tests. Wait for a few minutes (most of our tests run in
+<15 minutes). If the checks failed:
+
```
Screenshot of `convert to draft` link might also help (in case user opens a regular PR)  by mistake

edublancas replied:
added

neelasha23 commented:
Added some comments. Do you think it would help if there's a simple flowchart diagram to help understand which tutorial to refer in which scenario? common scenarios maybe `reviewing code` (then check out `Reviewing PRs`, `Maintainer's Guide`..)
@edublancas

yafimvo commented on contributing/submitting-pr.md:
```diff
@@ -202,7 +247,49 @@ Then ensure the menu displays the original repository (`ploomber/PROJECT_NAME`)
+
+Check the logs and fix them (see the [troubleshooting](troubleshooting.md) guide
+for tips). If you're unable to fix the issues after spending some time on them. You
+can message us on Slack so we can help you.
```
If you're unable to fix the issues after spending some time on them. You
can message us on Slack so we can help you.

I think it should be one sentence

yafimvo commented on contributing/submitting-pr.md:
```diff
@@ -202,7 +247,49 @@ Then ensure the menu displays the original repository (`ploomber/PROJECT_NAME`)
+
+Ensure there are no [merge conflicts](#fixing-merge-conflicts).
+
+If all the test pass and there are no merge conflicts,  you can [request a review](#requesting-a-review).
```
conflicts,  you can -> conflicts, you can (double space)

edublancas replied:
fixed

yafimvo reviewed (changes requested):
minor typos

edublancas commented on contributing/review.md:
```diff
@@ -1,7 +1,39 @@
-3. Share concise but direct feedback: strike a good balance between a short yet understandable message
\ No newline at end of file
+The first review should be the most detailed one, so upcoming reviews
+become smaller and take less time. The only exception is when there's something
```
wdym? in which case?

neelasha23 replied:
`The only exception is when there's something..` : the above points are also exceptions

edublancas replied:
I think the only case when it's fine to "overlook" a first review is when the PR is broken. if more test cases are needed or some refactoring is needed, you can still provide a detailed first review.

neelasha23 replied:
Ok. I think I misunderstood the sentence then . I meant second review doesn't necessarily have to be smaller than first for the above cases. First review can always be the extensive

edublancas commented on contributing/process-outline.md:
```diff
@@ -1,23 +1,22 @@
-you can choose one that you're interested in working on.
+## 1. Getting issue assigned
+
+The project's maintainers may assign you an issue, or  you can choose one that you're
```
fixed

edublancas commented:
thanks for the feedback, addressed all the comments!

> Do you think it would help if there's a simple flowchart diagram to help understand which tutorial

@neelasha23: we have some quick access links in the [home page](https://ploomber-contributing.readthedocs.io/en/latest/intro.html). do you think this can help? Or are you thinking that we can include an actual diagram (perhaps in the outline section?). adding the diagram would be simple with mermaid: https://mermaid.js.org/intro/

edublancas commented:
@neelasha23 thanks for the heads up on the "resolve conversation" button. I fixed github permissions (I'll add a little note so people can notify us if they're lacking permissions). please resolve the conversations if you're ok with the changes. I'll open a new issue with the flowchart

neelasha23 reviewed (approved):
Added a comment on typos. Looks good otherwise!
